camera_calibration/src/camera_calibration/calibrator.py

```diff
@@ -647,19 +666,25 @@ def cal_fromcorners(self, good):
 
         ipts = [ points for (points, _) in good ]
         opts = self.mk_object_points(boards)
-
         # If FIX_ASPECT_RATIO flag set, enforce focal lengths have 1/1 ratio
         intrinsics_in = numpy.eye(3, dtype=numpy.float64)
-        reproj_err, self.intrinsics, dist_coeffs, rvecs, tvecs = cv2.calibrateCamera(
-                   opts, ipts,
-                   self.size,
-                   intrinsics_in,
-                   None,
-                   flags = self.calib_flags)
-        # OpenCV returns more than 8 coefficients (the additional ones all zeros) when CALIB_RATIONAL_MODEL is set.
-        # The extra ones include e.g. thin prism coefficients, which we are not interested in.
-        self.distortion = dist_coeffs.flat[:8].reshape(-1, 1)
 
+        if self.camera_model == CAMERA_MODEL.PINHOLE:
+            reproj_err, self.intrinsics, dist_coeffs, rvecs, tvecs = cv2.calibrateCamera(
+                       opts, ipts,
+                       self.size,
+                       intrinsics_in,
+                       None,
+                       flags = self.calib_flags)
+            # OpenCV returns more than 8 coefficients (the additional ones all zeros) when CALIB_RATIONAL_MODEL is set.
+            # The extra ones include e.g. thin prism coefficients, which we are not interested in.
+            self.distortion = dist_coeffs.flat[:8].reshape(-1, 1)
+        elif self.camera_model == CAMERA_MODEL.FISHEYE:
+            reproj_err, self.intrinsics, self.distortion, rvecs, tvecs = cv2.fisheye.calibrate(
+                opts, ipts, self.size,
+                intrinsics_in, None, flags = self.fisheye_calib_flags)
+        else:
+            print("Something went wrong when selecting a model")
```

Comment:
Instead of smattering this warning around everywhere, I would prefer that we guarantee that self.camera_model is valid when the parameter is set, otherwise exit with an error.

Reply:
Also, I think the "more Pythonic" way is to ask for forgiveness rather than permission (AFRTP) instead of look before you leap (LBYL).

Reply:
It is guaranteed that self.camera_model is valid. That comment is not strictcly necessary for production and will be removed.